### PR TITLE
Dev/asolovev add comm profiling

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**
## Description

_Add a comprehensive description of proposed changes_

_List associated issue number(s) if exist(s): #6 (for example)_

_Documentation PR (if needed): #1340 (for example)_

_Benchmarks PR (if needed): https://github.com/IntelPython/scikit-learn_bench/pull/155 (for example)_

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.
- [ ] I have provided justification why quality metrics have changed or why changes are not expected.
- [ ] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
@Alexandr-Solovev I get a segmentation fault after the first run when running some algorithms repeatedly. Example code that segfaults after a few iterations:
```python
import numpy as np
import warnings
from sklearnex.linear_model import LinearRegression

rng = np.random.default_rng(seed=123)
m = 50_000
n = 100
for rep in range(5):
    X = rng.random(size=(m,n))
    y = rng.random(size=m)
    with warnings.catch_warnings():
        warnings.simplefilter("ignore", FutureWarning)
        LinearRegression(fit_intercept=False).fit(X, y)
```
The segfault is not encountered when using the main branch of oneDAL.

**Comment by david-cortes-intel** [comment]
@Alexandr-Solovev After this PR, would building with makefile option `REQPROFILE=yes` make any difference?

**Comment by david-cortes-intel** [comment]
@Alexandr-Solovev If this commit gets merged into the main branch, it then fails to compile sklearnex with the resulting library:
```
/localdisk2/mkl/dcortes/repos/oneDAL/__release_lnx/daal/latest/include/oneapi/dal/algo/pca/infer_types.hpp:82:19: error: static assertion failed due to requirement 'detail::is_valid_task_v<__va_list_tag[1]>'
   82 |     static_assert(detail::is_valid_task_v<Task>);
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
